pyvista/core/composite.py

```diff
@@ -505,6 +516,256 @@ def _recursive_iterator(
+        >>> multi.user_dict = data
+        >>> root.move_nested_field_data_to_root()
+
+        Check the root's user-dict. By default, the key names are preserved and the root
+        dict is updated with the nested dict.
+
+        >>> root.user_dict
+        {"foo": "bar"}
+
+        Clear the field data and re-add data to the nested user-dict.
+
+        >>> root.clear_field_data()
+        >>> multi.user_dict = data
+
+        Move the data again but use the ``'flat'`` mode. This time, a new key is added
+        which matches the nested block's name.
+
+        >>> root.move_nested_field_data_to_root(user_dict_mode='flat')
+        >>> root.user_dict
+        {"Block-00": {"foo": "bar"}}
+
+        """
+        _validation.check_contains(
+            ['prepend', 'preserve'], must_contain=field_data_mode, name='field_data_mode'
+        )
+        _validation.check_contains(
+            ['prepend', 'preserve', 'flat', 'nested'],
+            must_contain=user_dict_mode,
+            name='user_dict_mode',
+        )
+
+        root_field_data = self.field_data
+        prepend_names = field_data_mode == 'prepend'
+
+        iterator = self.recursive_iterator(
+            'all', node_type='parent', prepend_names=True, separator=separator
+        )
+        typed_iterator = cast(Iterator[tuple[tuple[int, ...], str, MultiBlock]], iterator)
+
+        for index, block_name, nested_multi in typed_iterator:
+            # Get nested field data to be moved
+            nested_field_data = nested_multi.field_data
+            if prepend_names:
+                # Add the field data to a temp mesh so we can rename the arrays
+                temp_mesh = pyvista.ImageData()
+                temp_field_data = temp_mesh.field_data
+                for old_name in nested_field_data:
+                    new_name = f'{block_name}{separator}{old_name}'
+                    temp_field_data[new_name] = nested_field_data[old_name]
+                field_data_to_copy = temp_field_data
+            else:
+                field_data_to_copy = nested_field_data
+
+            for array_name in field_data_to_copy:
+                # Check for nested user-dict data
+                if array_name.endswith(USER_DICT_KEY):
+                    root_user_dict = self.user_dict
+
+                    def raise_key_error(
+                        duplicate_key: str, block_name_: str, index_: tuple[int, ...]
+                    ) -> NoReturn:
+                        index_fmt = _format_nested_index(index_)
+                        msg = (
+                            f"The root user dict cannot be updated with data from nested MultiBlock at index {index_fmt} with name '{block_name_}'.\n"
+                            f"The key '{duplicate_key}' already exists in the root user dict and would be overwritten."
+                        )
+                        raise ValueError(msg)
+
+                    if user_dict_mode in 'preserve':
```

Comment:
This doesn't seem right, should probably be:
```suggestion
                    if user_dict_mode == 'preserve':
```
Will need to double check the tests for this